.github/scripts/setup_spark_remote.sh

```diff
@@ -2,31 +2,36 @@
 
 set -xv
 
-mkdir -p $HOME/spark
-cd $HOME/spark
+mkdir -p "$HOME"/spark
+cd "$HOME"/spark || exit
 
 version=$(wget -O - https://dlcdn.apache.org/spark/ | grep 'href="spark' | grep -v 'preview' | sed 's:</a>:\n:g' | sed -n 's/.*>//p' | tr -d spark- | tr -d / | sort -r --version-sort | head -1)
 spark=spark-${version}-bin-hadoop3
-spark_connect=spark-connect_2.12
+spark_connect="spark-connect_2.12"
+
+mkdir "${spark}"
 
-mkdir ${spark}
-cd {spark}
 
 SERVER_SCRIPT=$HOME/spark/${spark}/sbin/start-connect-server.sh
 
 ## check the spark version already exist ,if not download the respective version
-if [ -f ${SERVER_SCRIPT} ];then
+if [ -f "${SERVER_SCRIPT}" ];then
   echo "Spark Version already exists"
 else
-  wget https://dlcdn.apache.org/spark/spark-${version}/${spark}.tgz
-  tar -xvf ${spark}.tgz
+  if [ -f "${spark}.tgz" ];then
+    echo "${spark}.tgz already exists"
+  else
+    wget "https://dlcdn.apache.org/spark/spark-${version}/${spark}.tgz"
+  fi
+  tar -xvf "${spark}.tgz"
 fi
```

Comment:
Should the script clean up the downloaded .tgz file after extraction?

Reply:
Keeping it to avoid downloading every time